Consul/Client_Results.cs

```diff
@@ -44,6 +44,24 @@ public ConsulResult(ConsulResult other)
     /// </summary>
     public class QueryResult : ConsulResult
     {
+        public enum CacheResult
+        {
+            UNDEFINED,
+            MISS,
+            HIT
+        }
+
+        /// <summary>
+        /// In all cases the HTTP `X-Cache` header is always set in the response to either `HIT` or `MISS` indicating whether the response was served from cache or not.
+        /// </summary>
+        public CacheResult XCache { get; set; }
```

Comment:
I think it is more idiomatic to make this nullable. Thus the "no value" concept will be represented by `null` instead of `0/UNDEFINED`.